numpy/lib/function_base.py

```diff
@@ -2999,10 +2999,15 @@ def blackman(M):
     >>> plt.show()
 
     """
+    # Ensures at least float64 via 0.0.  M should be an integer, but conversion
+    # to double is safe for a range.
+    values = np.array([0.0, M])
```

Comment:
Hmm, I note that M is not checked for integer.  Why not just `0.0 + M`?

Reply:
Yeah, I generalized this from the version with `b` where it finds the full dtype once including `b`, which seemed sensible.  Happy to do either, this avoids finding dtype in the `M` special path, but adding it there is just as easy e.g. via `ones_like`.

This code could use some more love...